src/views/teacherwaitingroom/teacherwaitingroom.jsx

```diff
@@ -11,8 +11,8 @@ const render = require('../../lib/render.jsx');
 require('./teacherwaitingroom.scss');
 
 class TeacherWaitingRoom extends React.Component {
-    componentWillReceiveProps (nextProps) {
-        if (nextProps.approved) {
+    componentDidUpdate () {
```

Comment:
need to confirm this works

Reply:
Added similar handling in constructor. I think this should actually be better than before, because we would not expect `componentWillReceiveProps` to ever be called!